tests/components/zha/test_light.py

```diff
@@ -1198,6 +1199,151 @@ async def test_transitions(
     assert eWeLink_state.attributes["max_mireds"] == 500
 
 
+@patch(
+    "zigpy.zcl.clusters.lighting.Color.request",
+    new=AsyncMock(return_value=[sentinel.data, zcl_f.Status.SUCCESS]),
+)
+@patch(
+    "zigpy.zcl.clusters.general.LevelControl.request",
+    new=AsyncMock(return_value=[sentinel.data, zcl_f.Status.SUCCESS]),
+)
+@patch(
+    "zigpy.zcl.clusters.general.OnOff.request",
+    new=AsyncMock(return_value=[sentinel.data, zcl_f.Status.SUCCESS]),
+)
+async def test_on_with_off_color(hass, device_light_1):
+    """Test turning on the light and sending color commands before on/level commands for supporting lights."""
+
+    device_1_entity_id = await find_entity_id(Platform.LIGHT, device_light_1, hass)
+    dev1_cluster_on_off = device_light_1.device.endpoints[1].on_off
+    dev1_cluster_level = device_light_1.device.endpoints[1].level
+    dev1_cluster_color = device_light_1.device.endpoints[1].light_color
+
+    # Execute_if_off will override the "enhanced turn on from an off-state" config option that's enabled here
+    dev1_cluster_color.PLUGGED_ATTR_READS = {
+        "options": lighting.Color.Options.Execute_if_off
+    }
+    update_attribute_cache(dev1_cluster_color)
+
+    # turn on via UI
+    dev1_cluster_on_off.request.reset_mock()
+    dev1_cluster_level.request.reset_mock()
+    dev1_cluster_color.request.reset_mock()
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        "turn_on",
+        {
+            "entity_id": device_1_entity_id,
+            "color_temp": 235,
+        },
+        blocking=True,
+    )
+
+    assert dev1_cluster_on_off.request.call_count == 1
+    assert dev1_cluster_on_off.request.await_count == 1
+    assert dev1_cluster_color.request.call_count == 1
+    assert dev1_cluster_color.request.await_count == 1
+    assert dev1_cluster_level.request.call_count == 0
+    assert dev1_cluster_level.request.await_count == 0
+
+    assert dev1_cluster_on_off.request.call_args_list[0] == call(
+        False,
+        dev1_cluster_on_off.commands_by_name["on"].id,
+        dev1_cluster_on_off.commands_by_name["on"].schema,
+        expect_reply=True,
+        manufacturer=None,
+        tries=1,
+        tsn=None,
+    )
+    assert dev1_cluster_color.request.call_args == call(
+        False,
+        dev1_cluster_color.commands_by_name["move_to_color_temp"].id,
+        dev1_cluster_color.commands_by_name["move_to_color_temp"].schema,
+        color_temp_mireds=235,
+        transition_time=0,
+        expect_reply=True,
+        manufacturer=None,
+        tries=1,
+        tsn=None,
+    )
```

Comment:
It's not too important here, but how would I check the message order? (messages are sent on different clusters)
(That code doesn't care whether "on" or "move_to_color_temp" is sent first)